### ISSUE TITLE
Pawn Library has stopped working randomly (it seems)

### ISSUE THREAD

**Issue by rt-2** [pawn-lang/compiler#316]
**Is this a BUG REPORT, FEATURE REQUEST or QUESTION?**:

* [x] Bug Report
* [ ] Feature Request
* [ ] Question

**What happened**:
I get a "Pawn Library has stopped working" sometimes.

**What you expected to happen**:
It should compile OR give me an error.

**How to reproduce it (as minimally and precisely as possible)**:
I have not found a minimal way of reproducing it, I have had this bug for YEARS now, and it always seemed totally random. After recording the video I tried doing the same tests and now it works without problems, this is weird.
This is 2 videos as examples:
[Video 1](https://www.youtube.com/watch?v=_WaovhqukCQ);
[Video 2](https://www.youtube.com/watch?v=AM50URrA_EU);

**Anything else we need to know?**:
I'm kinda lost, maybe I'm not testing this the right way, enlighten me guys.

**Environment**:

* Operating System
Win7 64bit.
* Compiler version
3.10.7
* How are you invoking the compiler? Pawno, Sublime, vscode, sampctl or command-line?
Pawno
* If using sampctl, the version number

**Comment by Y-Less** [comment]
I've actually been chasing down a crash recently.  Might be the same one, might not, but this is a new one for me, not one that's been happening for years.

You need to comment out code and see if it still crashes.  If it does, remove more code (but make sure there are no errors, the crashes can be in the code gen stage, so errors will prevent the compiler reaching that stage).  Repeat until you have the smallest possible reproduction.

**Comment by rt-2** [comment]
I will definitely continue to investigate, but for now I have small clues only. It seems that the same code would work later without errors, I'm not sure of the factors but it seems to always happen on large files (I almost only have large files too), it seems to  be able to go away by commenting code that doesn't make error in other main files, and it seems to be able to go away only by adding empty lines.
I'm really not sure but it seems to have something to do with large file or certain sizes of files. Like if the compiler had an error handling the file and would read something distorted and crash, but that's really just a wild guess.

Thanks for the fast reply and for all your work.
I'll update this thread is I have any more infos.

**Comment by Y-Less** [comment]
Yes, I remember tracking down a similar crash once, where I would position my hand over F5 and ESC - one to start the compilation, the other to close the compiler window.  Then just spam the two and if I got to about 20 compiles in a row, I'd assume the crash was gone in the current code version.

**Comment by Jeroen52** [comment]
I haven't had the random compiler freezes/crashed on the Linux compiler yet.

However it is safe to say that this compiler was built specifically for the
docker image it runs on, I build code before I push it, and the working
directory (and entire system) is cleaned every separate build.

And the image is updated every month.

**Comment by Y-Less** [comment]
That doesn't really prove much.  The crashes are very rare, and only come up under very obscure circumstances, so it could just be that the code you're compiling has no issues, not that the Linux version is inherently more stable.

**Comment by Jeroen52** [comment]
That could be it, however I used to have plenty of compiler freezes a long while ago.

Just never in my recent setup.

**Comment by YashasSamaga** [comment]
@rt-2 were you using a debug build of the compiler? The debug version contains assertions which might help to isolate the problem.

**Comment by Y-Less** [comment]
No, run the compiler with a debug build.

**Comment by Y-Less** [comment]
@rt-2 I just released 3.10.8 which included a fix for one random crash, could you try that and report back please?

**Comment by rt-2** [comment]
I do not know of a way to reproduce the issue,
If it happens again I will report the issue here.
Thank you.
